### PR TITLE
Use Single Machine ID

### PR DESCRIPTION
**Pull request by corindwyer**
### Description of the Change
* Use a single machine id throughout the control plane
* Update the `SchedulingService.removeTask` API to hide the internal implementation details
* Start using `TASK_ATTRIBUTES_AGENT_INSTANCE_ID` consistently and remove `TASK_ATTRIBUTES_AGENT_ID` so it is not accidentally used

### CODE REVIEW

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/DefaultSchedulingService.java [hunk 1]
```diff
@@ -542,9 +551,25 @@ public void addTask(QueuableTask queuableTask) {
...
+    public void removeTask(String taskId) {
+        Optional<Pair<Job<?>, Task>> taskById = v3JobOperations.findTaskById(taskId);
+        if (taskById.isPresent()) {
+            Pair<Job<?>, Task> jobTaskPair = taskById.get();
+            Job job = jobTaskPair.getLeft();
+            Task task = jobTaskPair.getRight();
+            Pair<Tier, String> tierAssignment = JobManagerUtil.getTierAssignment(job, capacityGroupService);
+            QAttributes qAttributes = new V3QAttributes(tierAssignment.getLeft().ordinal(), tierAssignment.getRight());
+            // host name should be null if it doesn't exist for fenzo to not try to unassign it
+            String hostname = kubeIntegrationEnabled ?
```
Comment (tbak) [inline titus-server-master/src/main/java/com/netflix/titus/master/scheduler/DefaultSchedulingService.java]: We should format this as
```
<condition>
? a
: b;
```

File: titus-testkit/src/main/java/com/netflix/titus/testkit/model/job/JobComponentStub.java [hunk 2]
```diff
@@ -193,7 +193,7 @@ public void place(String taskId, AgentInstance agentInstance) {
                                 .withTimestamp(clock.wallTime())
                                 .build()
                         )
-                        .addToTaskContext(TaskAttributes.TASK_ATTRIBUTES_AGENT_ID, agentInstance.getId())
+                        .addToTaskContext("agent.id", agentInstance.getId())
```
Comment (tbak) [inline titus-testkit/src/main/java/com/netflix/titus/testkit/model/job/JobComponentStub.java]: Can we remove that?

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/DefaultSchedulingService.java (hunk 1 above)
Comment (corindwyer) [reply titus-server-master/src/main/java/com/netflix/titus/master/scheduler/DefaultSchedulingService.java]: Fixed

File: titus-testkit/src/main/java/com/netflix/titus/testkit/model/job/JobComponentStub.java (hunk 2 above)
Comment (corindwyer) [reply titus-testkit/src/main/java/com/netflix/titus/testkit/model/job/JobComponentStub.java]: Yes; fixed

**Pull request merged**